### PR TITLE
Me: ES6ify the Profile Gravatar component

### PR DESCRIPTION
**Pull request by tyxla**
This PR ES6ifies the Profile Gravatar component:
* Replacing `require` with `import`
* Replacing `module.exports` with `export default`
* Replacing `var` with `let` or `const`
* Using method shorthand
* Using template literals

This component is displayed in the top of the left sidebar in `/me`.

/cc @roccotripaldi @ryelle 

Test live: https://calypso.live/?branch=update/me-profile-gravatar-es6ify

### CODE REVIEW

File: client/me/profile-gravatar/index.jsx [hunk 1]
```diff
@@ -1,27 +1,28 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	debug = require( 'debug' )( 'calypso:me:sidebar-gravatar' );
+import React from 'react';
+import _debug from 'debug';
```
Comment (retrofox) [inline client/me/profile-gravatar/index.jsx]: Generally we use `debugFactory` for this lib.

File: client/me/profile-gravatar/index.jsx [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
-	componentDidMount: function() {
+	componentDidMount() {
 		debug( 'The ProfileGravatar component is mounted.' );
 	},
 
-	render: function() {
-		var profileURL = '//gravatar.com/' + this.props.user.username;
+	render() {
+		const profileURL = '//gravatar.com/' + this.props.user.username;
```
Comment (retrofox) [inline client/me/profile-gravatar/index.jsx]: you would use string templates ...

```es6
const profileURL = `//gravatar.com/${ this.props.user.username }`;
```

... but it's just my suggestion. It's up to you.

**Comment by tyxla** [comment]
Thanks for the feedback @retrofox 👍

File: client/me/profile-gravatar/index.jsx [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
+const debug = debugFactory( 'calypso:me:sidebar-gravatar' );
 
+export default React.createClass( {
 	displayName: 'ProfileGravatar',
 
 	mixins: [ eventRecorder ],
 
-	componentDidMount: function() {
+	componentDidMount() {
 		debug( 'The ProfileGravatar component is mounted.' );
```
Comment (retrofox) [inline client/me/profile-gravatar/index.jsx]: I know it isn't part of the goal of this PR but I wonder if does it make sense?
Removing this debug line allows us remove the `componentDidMount` method, the `debug` instance and finally get rid of importing `debug` component.

Again ... it's up to you. :-D

Comment (tyxla) [inline client/me/profile-gravatar/index.jsx]: I know that can be bugging, but let's leave it for another time - it's really not within the purpose of this PR, and there are quite a few components that we can do it for.

**Comment by roccotripaldi** [comment]
🚢

**Pull request merged**